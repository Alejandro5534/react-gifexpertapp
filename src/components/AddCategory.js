import { useState } from 'react'
import PropTypes from 'prop-types';
export const AddCategory = ({setCategories}) => {
    
    const [inputValue, setinputValue] = useState('')
    const handleInputChange = (e)=> {
        setinputValue(e.target.value);
    }
    const handleSubmit = (e) => {
        e.preventDefault();
        if (inputValue.trim().length>2) {
            setCategories(categories => [ inputValue,...categories])
            setinputValue('')
        }
        // console.log('Submit Hecho')
    }
    return (
          
            <form onSubmit = {handleSubmit}>    
                <input 
                    placeholder= "Escribe una o mas palabras clave para que encuentres, entre 10 GIFS, tu preferido"
                    type="text"
                    value = {inputValue}
                    onChange={handleInputChange}

                />
            </form>    
        
    )
}

AddCategory.prototype={
    setCategories: PropTypes.func.isRequired        
}