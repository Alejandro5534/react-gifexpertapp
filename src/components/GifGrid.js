// import { useState, useEffect } from "react";
// import { getGifs } from "../helpers/getGifs";
import { useFetchGifs } from "../hooks/useFetchGifs";
import { GifGridItem } from "./GifGridItem";

export const GifGrid = ({category}) => {
    // const [counter, setcounter] = useState(0);
    // const [images, setimages] = useState([])
    const {data:images,loading} = useFetchGifs(category);
    
    
    
    return(
        <>
        <h3 className="animate__animated animate__fadeIn"> {category}</h3>
        {loading && <p className="animate__animated animate__flash">Cargando..</p> }
        <div className="card-grid">
            
                {
                    images.map((img)=>(
                        <GifGridItem
                            key={img.id} 
                            {...img}
                        />    
                    ))
                }
            
            
        </div>
        </>
    );
}

