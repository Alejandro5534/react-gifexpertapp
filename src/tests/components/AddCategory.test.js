import { shallow } from "enzyme"
import { AddCategory } from "../../components/AddCategory"

describe('Pruebas en <AddCategory', () => {
    const setCategories = () => {}
    const wrapper = shallow(<AddCategory setCategories= {setCategories}  />)
    test('debe de msotrarse correctament', () => {
        expect(wrapper).toMatchSnapshot();
    })
    test('debe de cambiar la caja de texto', () => {
        const input = wrapper.find('input')
        const value = 'Hola mundo'
        input.simulate('change', { target:{ value: value} })
    })
    
})
                        