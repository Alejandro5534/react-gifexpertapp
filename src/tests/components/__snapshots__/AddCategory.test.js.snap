// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <AddCategory debe de msotrarse correctament 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    onChange={[Function]}
    placeholder="Escribe una o mas palabras clave para que encuentres, entre 10 GIFS, tu preferido"
    type="text"
    value=""
  />
</form>
`;
